src/components/CodeEditor.tsx

```diff
@@ -323,6 +328,10 @@ export const CodeEditor = ({
     }
   }, [!isStreaming, currentFile, code !== ""])
 
+  useWarnUserForUnsavedChanges({
+    hasUnsavedChanges: hasUnsavedChanges ?? false,
+  })
```

Comment:
why does this have to be in the CodeEditor, can you move it to CodeAndPreview? This file is already very large and I don't see why this functionality has to be in here

Reply:
Yes it can also be moved in CodeAndPreview and it will work fine.